### ISSUE TITLE
Sharing doesn't work after upgrading to 12.0

### ISSUE THREAD

**Issue by wojciechszela** [nextcloud/server#5247]
Since upgrading to 12.0 sharing feature is not visible for user. OCC shows the app as disabled. Re-enabling files_sharing and files_external apps fails with the following error

 [Exception]
  App "" cannot be installed because appinfo file cannot be read.

### Steps to reproduce
1. upgrade from 11 to 12.0
2. login as regular user and try to share any folder - notice you can not share anything, no such feature
3. on server run sudo -u www-data php occ app:list
4. notice that files_sharing and files_external are disabled even though in previous version were enabled
5. on server run sudo -u www-data php occ app:enable files_external
6. above command fails to enable the app with error:  App "" cannot be installed because appinfo file cannot be read.

### Expected behaviour
Files sharing apps should work after upgrade from version 11 to 12. Enabling/disabling files_sharing and files_external should result in enabling/disabling them.

### Actual behaviour
Tell us what happens instead

### Server configuration

**Operating system**: Ubuntu 16.10

**Web server:** Nginx 1.13

**Database:** postgresql 9.5

**PHP version:** php 7.0.18

**Nextcloud version:** 12.0.0

**Updated from an older Nextcloud/ownCloud or fresh install:** update from 11.0.3

**Where did you install Nextcloud from:**

**Signing status:**
<details>
<summary>Signing status</summary>

No errors have been found.
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

Enabled:
  - activity: 2.5.2
  - admin_audit: 1.2.0
  - bruteforcesettings: 1.0.2
  - calendar: 1.5.3
  - comments: 1.2.0
  - contacts: 1.5.3
  - dav: 1.3.0
  - encryption: 1.6.0
  - federatedfilesharing: 1.2.0
  - federation: 1.2.0
  - files: 1.7.2
  - files_pdfviewer: 1.1.1
  - files_texteditor: 2.4.1
  - files_trashbin: 1.2.0
  - files_videoplayer: 1.1.0
  - firstrunwizard: 2.1
  - gallery: 17.0.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.0.0
  - nextcloud_announcements: 1.1
  - notifications: 2.0.0
  - oauth2: 1.0.5
  - password_policy: 1.2.2
  - provisioning_api: 1.2.0
  - serverinfo: 1.2.0
  - sharebymail: 1.2.0
  - socialsharing_diaspora: 1.0.1
  - socialsharing_email: 1.0.1
  - socialsharing_facebook: 1.0.1
  - socialsharing_googleplus: 1.0.1
  - socialsharing_twitter: 1.0.1
  - spreed: 2.0.1
  - survey_client: 1.0.0
  - systemtags: 1.2.0
  - theming: 1.3.0
  - twofactor_backupcodes: 1.1.1
  - updatenotification: 1.2.0
  - workflowengine: 1.2.0
Disabled:
  - files_external
  - files_sharing
  - files_versions
  - user_external
  - user_ldap
</details>

**Are you using external storage, if yes which one:** local

**Are you using encryption:** yes

**Are you using an external user-backend, if yes which one:** no

#### LDAP configuration (delete this part if not used)
<details>
<summary>LDAP config</summary>

</details>

### Client configuration
**Browser:** any

**Operating system:** any

#### Web server error log
<details>
<summary>Web server error log</summary>

n/a
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

{"reqId":"UMbD98gQEP0GODYBvh7g","level":3,"time":"2017-06-04T22:04:48+00:00","remoteAddr":"172.113.248.211","user":"wojciech","app":"no app in context","method":"PROPFIND","url":"\/remote.php\/webdav\/","message":"Exception: {\"Exception\":\"Error\",\"Message\":\"Class 'OCA\\\\Files_Sharing\\\\Helper' not found\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/app.php(209): require_once()\\n#1 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/app.php(149): OC_App::requireAppFile('files_sharing')\\n#2 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/app.php(124): OC_App::loadApp('files_sharing')\\n#3 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/util.php(185): OC_App::loadApps(Array)\\n#4 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Files\\\/Filesystem.php(352): OC_Util::setupFS()\\n#5 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Files\\\/View.php(1123): OC\\\\Files\\\\Filesystem::resolvePath('\\\/files_encrypti...')\\n#6 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Files\\\/View.php(532): OC\\\\Files\\\\View->basicOperation('file_exists', '\\\/files_encrypti...')\\n#7 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Encryption\\\/Keys\\\/Storage.php(209): OC\\\\Files\\\\View->file_exists('\\\/files_encrypti...')\\n#8 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Encryption\\\/Keys\\\/Storage.php(107): OC\\\\Encryption\\\\Keys\\\\Storage->getKey('\\\/files_encrypti...')\\n#9 \\\/var\\\/www\\\/owncloud\\\/apps\\\/encryption\\\/lib\\\/KeyManager.php(569): OC\\\\Encryption\\\\Keys\\\\Storage->getSystemUserKey('pubShare_b32374...', 'OC_DEFAULT_MODU...')\\n#10 \\\/var\\\/www\\\/owncloud\\\/apps\\\/encryption\\\/lib\\\/KeyManager.php(157): OCA\\\\Encryption\\\\KeyManager->getPublicShareKey()\\n#11 \\\/var\\\/www\\\/owncloud\\\/apps\\\/encryption\\\/lib\\\/Users\\\/Setup.php(86): OCA\\\\Encryption\\\\KeyManager->validateShareKey()\\n#12 \\\/var\\\/www\\\/owncloud\\\/apps\\\/encryption\\\/lib\\\/AppInfo\\\/Application.php(73): OCA\\\\Encryption\\\\Users\\\\Setup->setupSystem()\\n#13 \\\/var\\\/www\\\/owncloud\\\/apps\\\/encryption\\\/appinfo\\\/app.php(30): OCA\\\\Encryption\\\\AppInfo\\\\Application->__construct(Array, true)\\n#14 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/app.php(209): require_once('\\\/var\\\/www\\\/ownclo...')\\n#15 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/app.php(149): OC_App::requireAppFile('encryption')\\n#16 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/app.php(124): OC_App::loadApp('encryption')\\n#17 \\\/var\\\/www\\\/owncloud\\\/remote.php(147): OC_App::loadApps(Array)\\n#18 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/apps\\\/files_sharing\\\/appinfo\\\/app.php\",\"Line\":29}","userAgent":"Mozilla\/5.0 (Windows) mirall\/2.1.0","version":"12.0.0.29"}
</details>

#### Browser log
<details>
<summary>Browser log</summary>

n/a
</details>

**Comment by schiessle** [comment]
please check if following file exists and if the webserver has read permissions: apps/files_sharing/appinfo/info.xml

**Comment by wojciechszela** [comment]
The file does exist and is readable by the proper user. The contents seem to be right.

I did upgrade manually, following these instructions https://docs.nextcloud.com/server/12/admin_manual/maintenance/manual_upgrade.html

Let me know please if I can provide any other helpful info.

**Comment by tlueder** [comment]
Hi There,

had the same problem.
I updated the cloud from 11.0.3 to 12.0.1 manually like I always do.
My fix was to delete all lines in the <description> </description> part of the info.xml

Had the same problem after upgrading the calender plugin and fixed it the same way.

To me it seems to like charset problem

**Comment by xydac** [comment]
Any update on this, It's still not working and giving error 
`App "" cannot be installed because appinfo file cannot be read.`

Files is there and readable with same permission as other apps.

I further downloaded it from git and replace the folder but it's still giving same error on enabling.

**Comment by avilkin** [comment]
Hm. I have the same problem after upgrading from 11 to 12.
I removed second and third strings from "description" tag in info.xml file, and, after which "file-sharing" was enabled without any errors. After that I had trying delete only "line breaks" in "description" and got error again. And finally, I reduced the number of characters (just deleting last sentence "More information...") and error gone.
Its strange, but seems like "description" must have <700 chars.
